### PR TITLE
Error codes using golang

### PR DESCRIPTION
**Pull request by wtrocki**
Creating for early feedback, but I have doubts if there is any benefit of having this done in golang.

As for now I do not see any obvious benefits of using GO for that purpose vs Node.js, however I'm happy to finish that.

Reasons: 

- There is no actual difference or benefit of migrating from js to go. Nothing blocking us etc.
- We need to duplicate efforts and maintenance rather than reusing similar codebase across SDKs.
- Suggested solution in https://github.com/redhat-developer/app-services-sdk-go/pull/343 just works and it is full separate job/implementation that can be tweaked 
- Go requires us to build structures and templates that overly complicate entire implementation and further maintenance.
- Build pipeline requires compilation and it is not using the branch files but "latest"

### CODE REVIEW

File: internal/apigen/cmd/api-generate-errors/main.go [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+	if repoErrorsPath == "" {
+		log.Fatalln("Missing required flag: --errors-folder")
+	}
+
+	clientConfig, err := metadata.GetClientMetadata(repoMetadataPath)
+	if err != nil {
+		log.Fatalln("no metadata found:", err)
+	}
+
+	for id, metadata := range *clientConfig {
```
Comment (wtrocki) [inline internal/apigen/cmd/api-generate-errors/main.go:37]: I just made code compile but this is not functional - it is missing error structs per API and reading json files.

**Comment by craicoverflow** [comment]
> We need to duplicate efforts and maintenance rather than reusing similar codebase across SDKs.

You could argue that if we did it all in Go in the first place (or an alternative language like Node.js) and centralized it, it would reduce maintenance and avoid duplication. Going with basic Bash/Node.js scripts means you need to copy the file across three (at least) different repos. If you need to fix something in one you need to update the others also.

> Build pipeline requires compilation and it is not using the branch files but "latest"

Not sure what the problem is with this. I think "latest" == HEAD. You can easily use the branch files if you prefer.

File: internal/apigen/cmd/api-generate-errors/main.go [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+		const (
+			{{range .Errors}}
+			// {{.Reason}}
+			ERROR_{{.ErrorID}} ServiceErrorCode = "{{.ErrorCode}}"
+			{{end}}
+		)
+		`)
+		data := struct {
+			Package string
+			Errors []interface
```
Comment (craicoverflow) [inline internal/apigen/cmd/api-generate-errors/main.go:59]:
```suggestion
			Errors []interface{}
```

**Comment by wtrocki** [comment]
I have tried to centralize it before golang cli was introduced in:

https://github.com/redhat-developer/app-services-sdk-go/pull/61

This ended up being overly complex to manage. We never copy 100% of the content - 
some adjustments are always needed.

**Comment by craicoverflow** [comment]
I'm happy with it if you prefer to keep it as Node.js and Bash - just know that this will mean to Go SDK automation now has a mix of three different languages.

**Comment by craicoverflow** [comment]
I have thought about it and I am more than happy to use the Bash/Node.js scripts as they are already working.

**Comment by wtrocki** [comment]
Closing for future reference

**Pull request closed without merging**